vllm/v1/worker/gpu_worker.py

```diff
@@ -209,6 +229,193 @@ def profile(self, is_start=True):
         else:
             self.profiler.stop()
 
+    def check_health(self) -> None:
+        # worker will always be healthy as long as it's running.
+        return
+
+
+@dataclass
+class WorkerProcHandle:
+    proc: BaseProcess
+    rank: int
+    ready_path: str
+    worker_response_mq: MessageQueue  # The worker process writes to this MQ
+
+
+class WorkerProc:
+    """Wrapper that runs one Worker in a separate process."""
+
+    READY_STR = "READY"
+
+    def __init__(
+        self,
+        vllm_config: VllmConfig,
+        local_rank: int,
+        rank: int,
+        distributed_init_method: str,
+        input_shm_handle: Handle,
+        ready_path: str,
+    ):
+        self.rank = rank
+        wrapper = WorkerWrapperBase(vllm_config=vllm_config)
+        wrapper.init_worker(vllm_config, local_rank, rank,
+                            distributed_init_method)
+        self.worker = wrapper.worker
+
+        pid = os.getpid()
+        _add_prefix(sys.stdout, f"VllmWorker rank={rank}", pid)
+        _add_prefix(sys.stderr, f"VllmWorker rank={rank}", pid)
+
+        # Initialize MessageQueue for receiving SchedulerOutput
+        self.worker_request_mq = MessageQueue.create_from_handle(
+            input_shm_handle, self.worker.rank)
+
+        # Initializes a message queue for sending the model output
+        self.worker_response_mq = MessageQueue(1, 1)
```

Comment:
we discussed in the slack that `MessageQueue` with a single reader might have minimal benefit and we can just use a `XSUB` from the executor side connected with `n` `PUB` from all workers. will that be implemented in this pr?

Reply:
I think we should land the current PR using shared memory message queues, and then I think there are a few different options we can explore